Feature/signin

JasonJeongjaeheon opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 데이터베이스 작업
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 - 해당 브랜치(PR)에서 구현하고자 하는 하나의 목표 작성
- signin 기능 구현

<br />

## :: 구현 사항 설명 - 해당 브랜치(PR)에서 작업한 내용 작성
- API 호출 시, bcrypt 암호화하여 signin 기능 구현
- signin 된 회원에게 jsonwebtoken 으로 token 발급
- signin 토큰 검증에 대한 기능은 아직 구현하지 않음


<br />

## :: 테스트 결과 이미지
1. Postman(Client Tool)을 이용한 API 테스트 결과 이미지
2. DB 작업 PR인 경우 Table 생성 및 수정 결과에 대해 확인할 수 있는 이미지

<br />

## :: 기타 질문 및 특이 사항
- 에러 핸들링 및 global error handling에 대해서 아직 헷갈립니다.
- layered pattern의 데이터 전체 흐름에 대해서 좀 더 구체적인 파악이 필요하다고 생각됩니다.(데이터 흐름 뿐만 아니라 데이터 타입은 어떻게 되는지 등)

Commit:
ADD:signin 기능 내 userService 구현 중

Commit:
MODIFY: userService authentication 기능 수정

Commit:
ADD:global error handling & signin bcryp compare를 위한 부분 수정

kyeongraekim commented on api/models/userDao.js:
```diff
@@ -0,0 +1,14 @@
+const getUserByEmail = async (email, password) => {
+    const user =
+        await AppData.query(
+        `SELECT * FROM 
+            customers c 
+            WHERE c.email = ?
+            `, [email]
+        )
```
select * 로 전체 선택이 아닌 개별 컬럼 선택을 통해서 어떠한 데이터들이 불러오는지 제 3의 개발자에게 가독성을 확보해주세요!~

kyeongraekim commented on api/controllers/userController.js:
```diff
@@ -0,0 +1,20 @@
+const { userService } = require('../services')
+
+const signIn = async (req, res) => {
```
controller에 wrapper 함수를 덧씌워서 비동기 에러시 해당 에러를 잡을 수 있게 수정해주세요!
기존 layered pattern 인증인가 모델 utills 파일에 참고할 만한 노드가 녹여져있을 겁니다!

Commit:
-error.js catchAsync module 생성
-userController.js catchAsync module 사용하여 error에 대하여 global error handler로 처리
-userDao.js Query 추출 데이터에 대한 가독성을 높이기 위하여 query 수정

kyeongraekim reviewed (approved):
Conflict 내용만 해결 부탁드릴게요!~

Commit:
FIX:signin branch conflict 해결

JasonJeongjaeheon commented:
conflict 해결 완료하였습니다!
